### PR TITLE
PMM-8990: fix build on Windows

### PR DESCRIPTION
**Pull request by ritbl**
- use generic syscall instead of depending on unix package
- fix CI for forked repos

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,6 +12,5 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/sirupsen/logrus v1.6.0
 	github.com/stretchr/testify v1.6.1
-	golang.org/x/sys v0.0.0-20200722175500-76b94024e4b6
```
Comment (ritbl) [inline go.mod:15]: not used

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+*.go text eol=lf
```
Comment (ritbl) [inline .gitattributes:1]: let's be explicit about encoding.
fixes running `goimports` on Windows

File: Makefile [hunk 3]
```diff
@@ -62,7 +62,7 @@ check-style:                    ## Run style checkers and linters.
 
 check-all: check check-style    ## Run all linters for new code..
 
-FILES = $(shell find . -type f -name '*.go')
+FILES = $(shell find . -type f -name "*.go")
```
Comment (ritbl) [inline Makefile:64]: fixes running make on Windows

File: Makefile [hunk 4]
```diff
@@ -28,10 +28,10 @@ release:                        ## Build pmm-admin release binary.
...
 
 init:                           ## Installs development tools
-	go build -modfile=tools/go.mod -o $(BIN_PATH)/go-consistent github.com/quasilyte/go-consistent
-	go build -modfile=tools/go.mod -o $(BIN_PATH)/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
-	go build -modfile=tools/go.mod -o $(BIN_PATH)/reviewdog github.com/reviewdog/reviewdog/cmd/reviewdog
-	go build -modfile=tools/go.mod -o $(BIN_PATH)/goimports golang.org/x/tools/cmd/goimports
+	go build -modfile=tools/go.mod -o "$(BIN_PATH)/go-consistent" github.com/quasilyte/go-consistent
+	go build -modfile=tools/go.mod -o "$(BIN_PATH)/golangci-lint" github.com/golangci/golangci-lint/cmd/golangci-lint
+	go build -modfile=tools/go.mod -o "$(BIN_PATH)/reviewdog" github.com/reviewdog/reviewdog/cmd/reviewdog
+	go build -modfile=tools/go.mod -o "$(BIN_PATH)/goimports" golang.org/x/tools/cmd/goimports
```
Comment (ritbl) [inline Makefile]: windows compatibility fix

File: Makefile [hunk 5]
```diff
@@ -75,5 +75,10 @@ env-down:                       ## Stop development environment.
...
 ci-reviewdog:                   ## Runs reviewdog checks.
-	$(BIN_PATH)/golangci-lint run -c=.golangci-required.yml --out-format=line-number | $(BIN_PATH)/reviewdog -f=golangci-lint -level=error -reporter=github-pr-check
-	$(BIN_PATH)/golangci-lint run -c=.golangci.yml --out-format=line-number | $(BIN_PATH)/reviewdog -f=golangci-lint -level=error -reporter=github-pr-review
+    ifneq ($(REVIEWDOG_GITHUB_API_TOKEN), )
+		$(BIN_PATH)/golangci-lint run -c=.golangci-required.yml --out-format=line-number | $(BIN_PATH)/reviewdog -f=golangci-lint -level=error -reporter=github-pr-check;
+		$(BIN_PATH)/golangci-lint run -c=.golangci.yml --out-format=line-number | $(BIN_PATH)/reviewdog -f=golangci-lint -level=error -reporter=github-pr-review;
+    else
+		echo "REVIEWDOG_GITHUB_API_TOKEN is not set, using golangci-lint without reviewdog"
+		$(BIN_PATH)/golangci-lint run -c=.golangci.yml --out-format=line-number --new-from-rev=main
+    endif
```
Comment (ritbl) [inline Makefile:82]: to fix CI for PRs from forked repos

File: main.go [hunk 6]
```diff
@@ -230,3 +223,12 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func handleInterrupts(signals chan os.Signal, cancel context.CancelFunc) {
```
Comment (ritbl) [inline main.go:228]: extracted to make code testable

File: Makefile [hunk 7]
```diff
@@ -28,10 +28,9 @@ release:                        ## Build pmm-admin release binary.
 	env CGO_ENABLED=0 go build -v $(LD_FLAGS) -o $(PMM_RELEASE_PATH)/pmm-admin
 
 init:                           ## Installs development tools
-	go build -modfile=tools/go.mod -o $(BIN_PATH)/go-consistent github.com/quasilyte/go-consistent
-	go build -modfile=tools/go.mod -o $(BIN_PATH)/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
-	go build -modfile=tools/go.mod -o $(BIN_PATH)/reviewdog github.com/reviewdog/reviewdog/cmd/reviewdog
-	go build -modfile=tools/go.mod -o $(BIN_PATH)/goimports golang.org/x/tools/cmd/goimports
+	go build -modfile=tools/go.mod -o "$(BIN_PATH)/go-consistent" github.com/quasilyte/go-consistent
+	go build -modfile=tools/go.mod -o "$(BIN_PATH)/golangci-lint" github.com/golangci/golangci-lint/cmd/golangci-lint
+	go build -modfile=tools/go.mod -o "$(BIN_PATH)/reviewdog" github.com/reviewdog/reviewdog/cmd/reviewdog
```
Comment (ritbl) [inline Makefile:33]: double quotes - to fix building on Windows

instead of `goimports` we can use `golangci-lint`

File: Makefile [hunk 8]
```diff
@@ -62,11 +61,10 @@ check-style:                    ## Run style checkers and linters.
...
 check-all: check check-style    ## Run all linters for new code..
 
-FILES = $(shell find . -type f -name '*.go')
+FILES = $(shell find . -type f -name "*.go")
 
-format:                         ## Format source code.
-	gofmt -w -s $(FILES)
-	$(BIN_PATH)/goimports -local github.com/percona/pmm-admin -l -w $(FILES)
+lint-fix:                         ## Format source code.
+	$(BIN_PATH)/golangci-lint run -c=.golangci.yml --fix --new-from-rev=main
```
Comment (ritbl) [inline Makefile:67]: `golangci-lint` includes formatting

**Comment by denisok** [comment]
@ritbl do you have the FB (feature build) ? Could you create one so we see test and build is ok

**Comment by ritbl** [comment]
@denisok right now FB cannot be built from forks. I submitted fix for [that](https://github.com/Percona-Lab/pmm-submodules/pull/2134), after it is merged. I will provide FB.

**Comment by denisok** [comment]
Thanks @ritbl . BTW there is old fashion way to do it, by changing subs directly: https://confluence.percona.com/display/PMM/Feature+Builds+-+Technical+design

**Comment by ritbl** [comment]
@denisok the old way - updating `.gitmodules`, right? if yes, unfortunately it doesn't work :(

**Pull request closed without merging**